src/mango_httpd.erl

```diff
@@ -80,6 +80,27 @@ handle_index_req(#httpd{method='POST', path_parts=[_, _]}=Req, Db) ->
     end,
 	chttpd:send_json(Req, {[{result, Status}, {id, Id}, {name, Name}]});
 
+%% Essentially we just iterate through the list of ddoc ids passed in and
+%% delete one by one. If an error occurs, all previous documents will be
+%% deleted, but an error will be thrown for the current ddoc id.
+handle_index_req(#httpd{method='POST', path_parts=[_, <<"_index">>,
+        <<"_bulk_delete">>]}=Req, Db) ->
+    {ok, Opts} = mango_opts:validate_bulk_delete(chttpd:json_body_obj(Req)),
+    Idxs = mango_idx:list(Db),
+    DDocs = get_bulk_delete_ddocs(Opts),
+    DelOpts = get_idx_w_opts(Opts),
+    lists:foreach(fun(DDocId0) ->
+        DDocId = convert_to_design_id(DDocId0),
+        Filt = fun(Idx) ->
+            IsDDoc = mango_idx:ddoc(Idx) == DDocId,
+            JsonType = mango_idx:type(Idx) == <<"json">>,
+            TextType = mango_idx:name(Idx) == <<"text">>,
+            IsDDoc andalso (JsonType orelse TextType)
+        end,
+        mango_idx:delete(Filt, Db, Idxs, DelOpts)
+    end, DDocs),
+    chttpd:send_json(Req, {[{result, <<"Indexes Deleted">>}]});
```

Comment:
Oh if you pass in  [ a, b , c ,d, e], and b and e fail, I would return "success": [a,c,d], "fail" : [b,e] (basically what you recommended).